src/legacy/ui/public/saved_objects/components/saved_object_save_modal.tsx

```diff
@@ -16,56 +16,148 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
-import React, { Fragment } from 'react';
-import PropTypes from 'prop-types';
-import { FormattedMessage } from '@kbn/i18n/react';
-
 import {
   EuiButton,
+  EuiCallOut,
   EuiFieldText,
+  EuiForm,
+  EuiFormRow,
   EuiModal,
   EuiModalBody,
   EuiModalFooter,
   EuiModalHeader,
   EuiModalHeaderTitle,
   EuiOverlayMask,
   EuiSpacer,
-  EuiCallOut,
-  EuiForm,
-  EuiFormRow,
   EuiSwitch,
 } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { Fragment } from 'react';
 
-export class SavedObjectSaveModal extends React.Component {
-  constructor(props) {
-    super(props);
+interface OnSaveProps {
+  newTitle: string;
+  newCopyOnSave: boolean;
+  isTitleDuplicateConfirmed: boolean;
+  onTitleDuplicate: () => void;
+}
 
-    this.state = {
-      title: props.title,
-      copyOnSave: false,
-      isTitleDuplicateConfirmed: false,
-      hasTitleDuplicate: false,
-      isLoading: false,
-    };
-  }
-  componentDidMount() {
-    this._isMounted = true;
-  }
+interface Props {
+  onSave: (arg0: OnSaveProps) => void;
+  onClose: () => void;
+  title: string;
+  showCopyOnSave: boolean;
+  objectType: string;
+  confirmButtonLabel?: any;
+  options?: any;
+}
+
+interface State {
+  title: string;
+  copyOnSave: boolean;
+  isTitleDuplicateConfirmed: boolean;
+  hasTitleDuplicate: boolean;
+  isLoading: boolean;
+}
+
+export class SavedObjectSaveModal extends React.Component<Props, State> {
+  public readonly state = {
+    title: this.props.title,
```

Comment:
This should not work properly, since `this.props` should still be empty at that point this is called. I think for setting the initial value from properties to state, we'll need doing this in the constructor (as earlier) (or going the full way with `getDerivedStateFromProps`.

Reply:
We've reviewed it and it can stay this way, cause it works:
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html